File: backend/shop-floor-assistance-backend/src/main/java/com/shopfloor/backend/api/transferobjects/ItemTO.java
package com.shopfloor.backend.api.transferobjects;

import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
public class ItemTO {

    private Long id;

    private String name;

    private String description;

    private Integer timeRequired;

    private Long createdBy;

    private Long updatedBy;

    private Date createdAt;

    private Date updatedAt;

    public ItemTO() {

    }

    public ItemTO(String name, String description, Integer timeRequired) {
        this.name = name;
        this.description = description;
        this.timeRequired = timeRequired;
    }
}

File: frontend/shop-floor-assistance-frontend/src/app/shared/types/itemTO.ts
export type itemTO = {
  id?: number; // Optional: Unique identifier for the item
  name: string; // Name of the item (required)
  description?: string; // Optional: Additional details or description of the item
  timeRequired?: number | null; // Optional: Estimated time required for the item (in minutes or seconds), can be null if not applicable
  createdBy?: number; // Optional: Identifier of the user who created the item
  updatedBy?: number; // Optional: Identifier of the user who last updated the item
  createdAt?: Date; // Optional: Timestamp indicating when the item was created
  updatedAt?: Date; // Optional: Timestamp indicating when the item was last updated
};
